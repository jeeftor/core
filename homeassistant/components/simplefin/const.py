"""Constants for the SimpleFIN integration."""
from __future__ import annotations

import logging
from typing import Final

DOMAIN: Final = "simplefin"

LOGGER = logging.getLogger(__package__)

CONF_ACCESS_URL = "access_url"

ICON_CHECKING = "mdi:checkbook"
ICON_CREDIT_CARD = "mdi:credit-card"
ICON_SAVINGS = "mdi:piggy-bank-outline"
ICON_INVESTMENT = "mdi:chart-areaspline"
ICON_UNKNOWN = "mdi:cash"
