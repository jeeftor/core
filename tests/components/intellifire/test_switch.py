"""Test intelliFire Sensors."""

from unittest.mock import patch

from freezegun import freeze_time
import pytest
from syrupy import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import setup_integration

from tests.common import MockConfigEntry, snapshot_platform


@freeze_time("2021-01-01T12:00:00Z")
@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_all_sensor_entities(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_config_entry_current: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    mock_fp,
) -> None:
    """Test all entities."""
    with (
        patch("homeassistant.components.intellifire.PLATFORMS", [Platform.SWITCH]),
        patch(
            "intellifire4py.unified_fireplace.UnifiedFireplace.build_fireplace_from_common",
            return_value=mock_fp,
        ),
    ):
        await setup_integration(hass, mock_config_entry_current)

        await snapshot_platform(
            hass, entity_registry, snapshot, mock_config_entry_current.entry_id
        )


async def test_switch(
    hass: HomeAssistant,
    mock_config_entry_current: MockConfigEntry,
    mock_fp,
) -> None:
    """Test if (config) switches get created."""
    with (
        patch("homeassistant.components.intellifire.PLATFORMS", [Platform.SWITCH]),
        patch(
            "intellifire4py.unified_fireplace.UnifiedFireplace.build_fireplace_from_common",
            return_value=mock_fp,
        ),
    ):
        await setup_integration(hass, mock_config_entry_current)

        test_entity = hass.states.get("switch.intellifire_flame")
        assert test_entity is not None
        assert test_entity.name == "IntelliFire Flame"
        assert test_entity.state == "on"
        assert test_entity.attributes == {
            "attribution": "Data provided by unpublished Intellifire API",
            "friendly_name": "IntelliFire Flame",
        }

        await hass.services.async_call(
            "switch",
            "turn_off",
            {"entity_id": test_entity.entity_id},
            blocking=True,
        )

        assert test_entity.state == "off"
