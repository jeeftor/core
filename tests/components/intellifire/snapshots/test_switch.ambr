# serializer version: 1
# name: test_all_sensor_entities[switch.intellifire_flame-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.intellifire_flame',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Flame',
    'platform': 'intellifire',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'flame',
    'unique_id': 'on_off_mock_serial',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_sensor_entities[switch.intellifire_flame-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by unpublished Intellifire API',
      'friendly_name': 'IntelliFire Flame',
    }),
    'context': <ANY>,
    'entity_id': 'switch.intellifire_flame',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_all_sensor_entities[switch.intellifire_pilot_light-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.intellifire_pilot_light',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Pilot light',
    'platform': 'intellifire',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'pilot_light',
    'unique_id': 'pilot_mock_serial',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_sensor_entities[switch.intellifire_pilot_light-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by unpublished Intellifire API',
      'friendly_name': 'IntelliFire Pilot light',
    }),
    'context': <ANY>,
    'entity_id': 'switch.intellifire_pilot_light',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
